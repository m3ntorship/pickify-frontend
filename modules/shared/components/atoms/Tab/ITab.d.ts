import type { IRadio } from '../Radio/IRadio';

declare namespace ITab {
  export interface IProps extends IData, IHandlers {
    disabled?: boolean;
    onlyLabel?: boolean;
  }

  export interface IData extends IRadio.IData {
    value: 'Image Poll' | 'Mini survey' | 'Text Poll';
    checkedValue: string;
    svg: JSX.Element;
  }

  export interface IHandlers {
    changeValHandler?: (e: React.ChangeEvent<HTMLInputElement>) => void;
  }
}
export { ITab };
