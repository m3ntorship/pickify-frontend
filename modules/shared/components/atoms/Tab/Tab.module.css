.tab-default {
  @apply cursor-pointer hover:bg-grey-shd6 flex items-center py-1.5 pr-3.5 pl-3 rounded-full;
}
.tab-checked {
  @apply bg-transparent md:bg-white border border-grey;
}
.tab-unchecked {
  @apply bg-grey-bg;
}

.tab-type {
  @apply text-sm hidden md:inline-block font-medium text-dark-grey;
}

.tab-disabled {
  @apply bg-grey-shd6 opacity-40 cursor-not-allowed border-0;
}

.icon {
  @apply md:mr-xs ml-xs;
}
