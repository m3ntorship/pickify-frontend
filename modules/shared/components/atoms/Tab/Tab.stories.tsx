import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import styles from './Tab.module.css';
import ImagePoll from '../../icons/imagePoll.svg';
import Tab from './Tab';
import type { ITab } from './ITab';
import { EPostType } from '../../../types/postFeed/EPostType';

export default {
  title: 'Atoms/Tab',
  component: Tab,
};

const Template: Story<ITab.IProps> = (args): ReactElement => <Tab {...args} />;
export const tab = Template.bind({});
tab.args = {
  value: EPostType.ImagePoll,
  id: 'anyid',
  svg: <ImagePoll className={styles.icon} />,
  checkedValue: 'image poll',
  disabled: false,
  onlyLabel: false,
};
