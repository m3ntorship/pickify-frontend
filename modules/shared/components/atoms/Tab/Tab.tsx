import React from 'react';
import classNames from 'classnames';
import type { FC, ReactElement } from 'react';
import styles from './Tab.module.css';
import Radio from '../Radio/index';
import type { ITab } from './ITab';

const Tab: FC<ITab.IProps> = (props): ReactElement => {
  const {
    value,
    id,
    svg,
    checkedValue,
    changeValHandler,
    disabled,
    onlyLabel,
  } = props;

  const tabClasses: string = classNames(styles['tab-default'], {
    [styles['tab-checked']]: value === checkedValue,
    [styles['tab-unchecked']]: value !== checkedValue,
    [styles['tab-disabled']]: disabled,
  });

  let content = 'Image Poll';

  if (value === 'image poll') {
    content = 'Image Poll';
  } else if (value === 'text poll') {
    content = 'Text Poll';
  } else {
    content = 'Mini survey';
  }

  return (
    <div className="flex">
      <label
        htmlFor={id}
        data-value={value}
        className={tabClasses}
        data-testid="tab"
      >
        <Radio
          size="small"
          value={value}
          name="tab groub"
          onChange={changeValHandler}
          id={id}
          disabled={disabled}
          defaultChecked={checkedValue === value}
          onlyLabel={onlyLabel}
        />
        {!onlyLabel && svg}
        <span className={styles['tab-type']}>{content}</span>
      </label>
    </div>
  );
};

export default Tab;
