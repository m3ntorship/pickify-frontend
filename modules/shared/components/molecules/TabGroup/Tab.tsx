import React, { useState } from 'react';
import classNames from 'classnames';
import type { FC, ReactElement } from 'react';
import styles from './TabGroup.module.css';
import Radio from '../../atoms/Radio/index';
import type { ITabGroup } from './types/ITabGroup';

const Tab: FC<ITabGroup.ITabProps> = (props): ReactElement => {
  const { active, id, svg, children } = props;

  const [checkedValue, setCheckedValue] = useState<string>('');

  const radioCheckedHandler = (
    e: React.ChangeEvent<HTMLInputElement>,
  ): void => {
    setCheckedValue(e.currentTarget.value);
  };

  const tabClasses: string = classNames(styles['tab-default'], {
    [styles['tab-active']]: active,
    [styles['tab-in-active']]: !active,
  });

  return (
    <label
      htmlFor={id}
      className={tabClasses}
      // onClick={toggleActiveHandler}
      data-testid="tab"
      // title={active && 'checkedTab'}
    >
      <Radio
        size="small"
        value={checkedValue}
        name="tab groub"
        onChange={radioCheckedHandler}
        id={id}
        defaultChecked={active}
      />
      {svg}
      <span className={styles['tab-type']}>{children}</span>
    </label>
  );
};

export default Tab;
