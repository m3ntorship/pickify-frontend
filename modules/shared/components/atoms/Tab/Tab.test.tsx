import React from 'react';
import * as renderer from 'react-test-renderer';
import ImagePoll from '../../icons/imagePoll.svg';
import Tab from './Tab';

describe('Tab', () => {
  it('should render tab with default status', () => {
    const changeValHandler = jest.fn();

    const tree = renderer
      .create(
        <Tab
          checkedValue="Image Poll"
          changeValHandler={changeValHandler}
          value="Mini survey"
          id="anyid"
          svg={<ImagePoll />}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render tab wit only label status', () => {
    const changeValHandler = jest.fn();

    const tree = renderer
      .create(
        <Tab
          checkedValue="Image Poll"
          changeValHandler={changeValHandler}
          value="Mini survey"
          id="anyid"
          svg={<ImagePoll />}
          onlyLabel
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render tab with disabled status', () => {
    const changeValHandler = jest.fn();

    const tree = renderer
      .create(
        <Tab
          checkedValue="Image Poll"
          changeValHandler={changeValHandler}
          value="Mini survey"
          id="anyid"
          svg={<ImagePoll />}
          disabled
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render tab with checked status', () => {
    const changeValHandler = jest.fn();

    const tree = renderer
      .create(
        <Tab
          checkedValue="Image Poll"
          changeValHandler={changeValHandler}
          value="Image Poll"
          id="anyid"
          svg={<ImagePoll />}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
