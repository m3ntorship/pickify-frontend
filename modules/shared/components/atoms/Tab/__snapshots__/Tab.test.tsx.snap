// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab should render tab wit only label status 1`] = `
<div
  className="flex"
>
  <label
    className="tab-default tab-unchecked"
    data-testid="tab"
    data-value="Mini survey"
    htmlFor="anyid"
  >
    <div
      className="radio_container"
    >
      <input
        className="radio_input size_small"
        data-testid="Radio"
        defaultChecked={false}
        disabled={false}
        id="anyid"
        name="tab groub"
        onChange={[MockFunction]}
        type="radio"
        value="Mini survey"
      />
      <div
        className="radio_background unchecked size_small checkmark_small only_label"
      />
    </div>
    <span
      className="tab-type"
    >
      Mini survey
    </span>
  </label>
</div>
`;

exports[`Tab should render tab with checked status 1`] = `
<div
  className="flex"
>
  <label
    className="tab-default tab-checked"
    data-testid="tab"
    data-value="Image Poll"
    htmlFor="anyid"
  >
    <div
      className="radio_container"
    >
      <input
        className="radio_input size_small"
        data-testid="Radio"
        defaultChecked={true}
        disabled={false}
        id="anyid"
        name="tab groub"
        onChange={[MockFunction]}
        type="radio"
        value="Image Poll"
      />
      <div
        className="radio_background unchecked size_small checkmark_small"
      />
    </div>
    <svg
      data-file-name="SvgImagePoll"
    />
    <span
      className="tab-type"
    >
      Mini survey
    </span>
  </label>
</div>
`;

exports[`Tab should render tab with default status 1`] = `
<div
  className="flex"
>
  <label
    className="tab-default tab-unchecked"
    data-testid="tab"
    data-value="Mini survey"
    htmlFor="anyid"
  >
    <div
      className="radio_container"
    >
      <input
        className="radio_input size_small"
        data-testid="Radio"
        defaultChecked={false}
        disabled={false}
        id="anyid"
        name="tab groub"
        onChange={[MockFunction]}
        type="radio"
        value="Mini survey"
      />
      <div
        className="radio_background unchecked size_small checkmark_small"
      />
    </div>
    <svg
      data-file-name="SvgImagePoll"
    />
    <span
      className="tab-type"
    >
      Mini survey
    </span>
  </label>
</div>
`;

exports[`Tab should render tab with disabled status 1`] = `
<div
  className="flex"
>
  <label
    className="tab-default tab-unchecked tab-disabled"
    data-testid="tab"
    data-value="Mini survey"
    htmlFor="anyid"
  >
    <div
      className="radio_container"
    >
      <input
        className="radio_input size_small"
        data-testid="Radio"
        defaultChecked={false}
        disabled={true}
        id="anyid"
        name="tab groub"
        onChange={[MockFunction]}
        type="radio"
        value="Mini survey"
      />
      <div
        className="radio_background unchecked size_small checkmark_small unchecked_disabled"
      />
    </div>
    <svg
      data-file-name="SvgImagePoll"
    />
    <span
      className="tab-type"
    >
      Mini survey
    </span>
  </label>
</div>
`;
